### PR TITLE
Replacement of bad_request with field-focused bad_form_request

### PR DESCRIPTION
**Pull request by szabozoltan69**

### CODE REVIEW

File: api/views.py
Comment (thenav56) [inline api/views.py:47]: @frozenhelium Is this format okay for the client-side?

**Review by thenav56 — APPROVED** [review]
LGTM

File: api/views.py
Comment (batpad) [reply api/views.py:47]: I don't think this is the exact format we want. Let's hold on merging until we can discuss this with @frozenhelium and @tnagorra - but, yes, hopefully that's just a matter of changing the JSON structure in this one function - otherwise this looks good!

Comment (tnagorra) [reply api/views.py:47]: This looks good for now.

**Review by batpad — APPROVED** [review]

**Comment by szabozoltan69** [comment]
Thanks @batpad @tnagorra @thenav56 @frozenhelium !
(I still miss that feature that at once multiple field can be "shown" as erroneous.)
Before merging, could you suggest me other .py files where the old style error messaging is still present?

**Comment by szabozoltan69** [comment]
We should change frontend side also, because the old code gives this (in case of wrong password):
image

**Comment by szabozoltan69** [comment]
Refers to https://github.com/IFRCGo/go-api/issues/1532

**Comment by batpad** [comment]
> We should change frontend side also, because the old code gives this (in case of wrong password):

Yes, of course 🤦 . Let's hold off on merging this until @frozenhelium or @tnagorra can say what it would take to get the frontend to work with this style? I guess ideally this would mean moving these forms over to the new forms? But not sure how much work that could be. We should probably hold off on merging the backend change before we have the frontend work lined up.

**Comment by szabozoltan69** [comment]
Frontend changes: https://github.com/IFRCGo/go-frontend/pull/2540

**Comment by frozenhelium** [comment]
@szabozoltan69 I think this also works for now, but in my opinion we should generalize the field names a bit since the error message might not always be a string. My suggestion for now is:
```js
error = [
  {
    field_name: <field_name>,

    // Could be string or object or array depending on type of field
    error: <error>,
  },
];
```

**Comment by szabozoltan69** [comment]
Thanks @frozenhelium – it's not quite clear how to create a generalized type in Python. (Or just simply we use the one that is appropriate at the different locations.)

**Comment by tnagorra** [comment]
The client also needs to know what the type of the field is. So, either we have to have an additional field to indicate if the field is `Array`, `Object` or `Literal` or we have three different fields for their errors like `arrayError`, `objectError` and `literalError`

**Pull request closed without merging**
